### ISSUE TITLE
Investigate the creation of a lambda function  to run the scripts that scans the status of the accounts

### ISSUE THREAD

**Issue by crochcunill** [bcgov/cloud-pathfinder#2316]
**Describe the Issue**
As a QA, I would like that anybody in the team can easily run the test scripts.
Currently, to run the script you need to download the code and run it from your machine. It also requires to supply the appropriate AWS credentials. This makes it difficult to run, specially for Live (currently the tester has to ask the SysOp to run it , as the tester does not have Live master credentials)

**Additional Context**
See the first iteration of this ticket for more information #2258

**Acceptance Criteria**
- Check if it is technically possible to run the python scripts as a lambda function without creating security risks
- Check with the team if it makes sense to implement such function (easy to trigger the function, function does not interfere with other artifacts, code control of the function....)
- If it goes ahead, create the ticket to port the test script to Lambda.

**Comment by crochcunill** [comment]
Invoke a Lambda from GitHub actions: https://github.com/marketplace/actions/invoke-aws-lambda

**Comment by crochcunill** [comment]
The posting https://github.com/marketplace/actions/invoke-aws-lambda suggest that is possible to invoke a lambda function from GitHub. Unfortunately, it has not work for me. I get the following error![image.png](https://images.zenhubusercontent.com/620eba1ac6937a4e70b34b15/cdb33831-f86c-4d8d-8936-91c543bc07ef)
I have also tried using OIDC authentication, but same problem. The values of the tokens (stored as GitHub secrets) do not validate the request. Obviously I have check and reckeck obtaining the same result.

In addition, I have realized that Lambdas are quite restricted in what they can and can't do. I use python scripts and import the _os_ module, to be able to run some _os_ commands. Only the most basic ones (_ls_,  _cd_...) work, but other like _cat_ do not. 

The scripts that I wanted to run as lambda make a heavy use of the _jq_ package, as it saves an enormous amount of work parsing key/value pairs. There is not a straightforward way to add it. Seems to be possible, but nobody that claims to have done it has posted a useful response or example on how to do it.

The last but not the least: the scripts I wanted to run in Lambda use the AWS CLI to extract information about the different accounts. As before, there is not an easy way to install the AWS CLI in a serverless Lambda (probably it goes against the same idea of having serverless functions). An option, is to use (in python) boto3. Basically all the commands in aws cli have a mirror in boto3. In this case, porting my scripts to lambda would imply redoing all the invocations by translating the api calls from using aws cli to boto3.

The end result is that the convenience of having (some, not all) the test script run in a CI/CD GitHub pipeline does not seem worth enough to justify the amount of work required to rewrite the current script as Lambda functions.
